core/src/main/java/com/linecorp/armeria/client/ClientCacheControl.java

```diff
@@ -0,0 +1,268 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static com.linecorp.armeria.internal.ArmeriaHttpUtil.parseCacheControl;
+import static com.linecorp.armeria.internal.ArmeriaHttpUtil.parseCacheControlSeconds;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.CacheControl;
+
+/**
+ * Directives for HTTP caching mechanisms in requests.
+ *
+ * @see ClientCacheControlBuilder
+ * @see <a href="https://developer.mozilla.org/docs/Web/HTTP/Headers/Cache-Control">Cache-Control (MDN)</a>
+ * @see <a href="https://stackoverflow.com/q/14541077">Why is Cache-Control header sent in a request?</a>
+ */
+public final class ClientCacheControl extends CacheControl {
+
+    /**
+     * An empty instance with all directives disabled.
+     */
+    public static final ClientCacheControl EMPTY = new ClientCacheControlBuilder().build();
+
+    /**
+     * {@code "no-cache"}.
+     */
+    public static final ClientCacheControl FORCE_NETWORK = new ClientCacheControlBuilder()
+            .noCache().build();
+
+    /**
+     * {@code "only-if-cached, max-stale=2147483647"}.
+     */
+    public static final ClientCacheControl FORCE_CACHE = new ClientCacheControlBuilder()
+            .onlyIfCached()
+            .maxStaleSeconds(Integer.MAX_VALUE)
+            .build();
+
+    /**
+     * Parses the specified {@code "cache-control"} header values into a {@link ClientCacheControl}.
+     * Note that any unknown directives will be ignored.
+     *
+     * @return the {@link ClientCacheControl} decoded from the specified header values.
+     */
+    public static ClientCacheControl parse(String... directives) {
+        return parse(ImmutableList.copyOf(requireNonNull(directives, "directives")));
+    }
+
+    /**
+     * Parses the specified {@code "cache-control"} header values into a {@link ClientCacheControl}.
+     * Note that any unknown directives will be ignored.
+     *
+     * @return the {@link ClientCacheControl} decoded from the specified header values.
+     */
+    public static ClientCacheControl parse(Iterable<String> directives) {
+        requireNonNull(directives, "directives");
+        final ClientCacheControlBuilder builder = new ClientCacheControlBuilder();
+        for (String d : directives) {
+            parseCacheControl(d, (name, value) -> {
+                switch (name) {
```

Comment:
Just curious. What if using `Map<String, BiConsumer<ClientCacheControlBuilder, String>>` here? Would it be better than `switch-case`? What do you think?

Reply:
Sounds good to me.